### PR TITLE
OCM-8063 | ci: Automated id:73610

### PR DESCRIPTION
**Pull request by jerichokeyne**
**What this PR does / why we need it**: Automates test case OCP-73610 which is about testing the default_ingress component_routes option

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (OCM-xxxx)*:
Fixes #[OCM-8063](https://issues.redhat.com//browse/OCM-8063)

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Build
- [x] CI
- [ ] Documentation
- [ ] Performance
- [x] Refactor
- [ ] Style
- [ ] Unit tests
- [ ] Subsystem tests

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: tests/e2e/classic_ingress_test.go [hunk 1]
```diff
@@ -80,6 +102,250 @@ var _ = Describe("Classic Ingress", ci.FeatureIngress, func() {
...
+			ingress, err := cms.RetrieveClusterIngress(ci.RHCSConnection, clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ingress.ComponentRoutes()["oauth"].Hostname()).To(Equal((*args.ComponentRoutes)["oauth"]["hostname"]))
+			Expect(ingress.ComponentRoutes()["oauth"].TlsSecretRef()).To(Equal((*args.ComponentRoutes)["oauth"]["tls_secret_ref"]))
+			Expect(ingress.ComponentRoutes()["console"].Hostname()).To(Equal((*args.ComponentRoutes)["console"]["hostname"]))
+			Expect(ingress.ComponentRoutes()["console"].TlsSecretRef()).To(Equal((*args.ComponentRoutes)["console"]["tls_secret_ref"]))
+			Expect(ingress.ComponentRoutes()["downloads"].Hostname()).To(Equal((*args.ComponentRoutes)["downloads"]["hostname"]))
+			Expect(ingress.ComponentRoutes()["downloads"].TlsSecretRef()).To(Equal((*args.ComponentRoutes)["downloads"]["tls_secret_ref"]))
+
+			By("update ingress component rountes")
```
Comment (xueli181114) [inline tests/e2e/classic_ingress_test.go:145]:
```suggestion
			By("update ingress component routes")
```

File: tests/tf-manifests/rhcs/ingresses/classic/main.tf [hunk 2]
```diff
@@ -20,6 +20,6 @@ resource "rhcs_default_ingress" "default_ingress" {
   cluster_routes_tls_secret_ref    = var.cluster_routes_tls_secret_ref
   id                               = var.id
   load_balancer_type               = var.load_balancer_type
-
+  component_routes = var.component_routes
```
Comment (xueli181114) [inline tests/tf-manifests/rhcs/ingresses/classic/main.tf:23]: run `make fmt` to format the manifest files

File: tests/e2e/classic_ingress_test.go [hunk 3]
```diff
@@ -80,6 +102,250 @@ var _ = Describe("Classic Ingress", ci.FeatureIngress, func() {
 			Expect(string(ingress.LoadBalancerType())).To(Equal("nlb"))
 		})
 
+	It("update ingress components - [id:73610]",
+		ci.Day2,
```
Comment (xueli181114) [inline tests/e2e/classic_ingress_test.go:106]: Split the validation checkpoints to another negative medium case.

Comment (jerichokeyne) [reply tests/e2e/classic_ingress_test.go:106]: @xueli181114 Sorry it took so long to get back to you. But, what label(s) should I be using for the negative test case? The only one I see for negative test cases is `Day1Negative`, is that what I should use?

Comment (jerichokeyne) [reply tests/e2e/classic_ingress_test.go:106]: @xueli181114 I split the negative tests into another test case: https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-75067. Can you take a look at it, as well as take another look at this PR?

Comment (xueli181114) [reply tests/e2e/classic_ingress_test.go:106]: @jerichokeyne The labels.Medium generally means negative validation cases.

File: tests/e2e/classic_ingress_test.go [hunk 4]
```diff
@@ -80,6 +102,255 @@ var _ = Describe("Classic Ingress", ci.FeatureIngress, func() {
...
+						"hostname":       "oauth.test.example.com",
+						"tls_secret_ref": "test-oauth",
+					},
+				},
+				Cluster: &clusterID,
+			}
+			output, err := ingressService.Apply(&args)
+			Expect(err).To(HaveOccurred())
+			output = strings.ReplaceAll(output, "\n", " ")
+			Expect(output).To(ContainSubstring("All component route kinds must be specified. Missing [console, downloads]"))
```
Comment (radtriste) [inline tests/e2e/classic_ingress_test.go:194]: You could use `helper.ExpectTFErrorContains` which is doing the same

**Review by radtriste — COMMENTED** [review]
One minor comment

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**